cache.go

```diff
@@ -153,11 +166,19 @@ func (c *Cache[K, V]) set(key K, value V, ttl time.Duration) *Item[K, V] {
 	}
 
 	// create a new item
-	item := NewItem(key, value, ttl, c.options.enableVersionTracking)
+	item := newItemWithOpts(key, value, ttl, c.options.itemOpts...)
 	elem = c.items.lru.PushFront(item)
 	c.items.values[key] = elem
 	c.updateExpirations(true, elem)
 
+	if c.options.maxCost != 0 {
```

Comment:
Just a comment: most probably the if clause can be removed (just retaining the contents of it). The addition does not have any performance impact and the for loop will not be executed anyway if maxCost is not set, as both the cost and maxCost will always be 0 in such case. What do you think?